.github/workflows/pytest.yml

```diff
@@ -0,0 +1,45 @@
+name: pytest (RHEL 8, RHEL 9)
+
+on:
+  pull_request:
+    paths:
+      - "src/*"
+
+jobs:
+  pytest:
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - name: "Fedora Rawhide"
+            image: "registry.fedoraproject.org/fedora:rawhide"
+          - name: "Fedora Latest"
+            image: "registry.fedoraproject.org/fedora:latest"
+          - name: "CentOS Stream 9"
+            image: "quay.io/centos/centos:stream9"
+          - name: "CentOS Stream 8"
+            image: "quay.io/centos/centos:stream8"
+
+    runs-on: "ubuntu-latest"
+    container:
+      image: ${{ matrix.image }}
+
+    steps:
+      - name: "Checkout the repository"
+        uses: actions/checkout@v4
+
+      - name: "Install dependencies"
+        # Setuptools is required because we still rely on 'distutils' which have been dropped in Python 3.12
+        run: |
+          dnf --setopt install_weak_deps=False install -y git-core python3-pip
+          python3 -m pip install --upgrade pip wheel
+          python3 -m pip install -r src/insights_client/tests/requirements.txt
+          python3 -m pip install git+https://github.com/RedHatInsights/insights-core.git
+          python3 -m pip install setuptools
+
+      - name: "Run pytest"
+        env:
+          PYTEST_ADDOPTS: "--color=yes --code-highlight=yes --showlocals"
+          PYTHONPATH: "src/"
+        run: |
+          python3 -m pytest src/insights_client/tests
```

Comment:
I notice that we have one test (`test_sed`) that is created at build time with meson; because of that, running `pytest` directly from the sources will ignore it.

So, as much as extra steps this looks like, I think we need to run the tests using meson, i.e.:
```bash
$ meson setup --prefix /usr build
$ meson test -C build -v
```
this means also installing `meson` when setting up things in this workflow